hoomd/mpcd/update.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2009-2024 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""MPCD updaters."""
+
+from . import _mpcd
+from hoomd.operation import Updater
+from hoomd.data.parameterdicts import ParameterDict
+from hoomd.logging import log
+
+
+class ReverseNonequilibriumShearFlow(Updater):
+    """Reverse Perturbation method to establish nonequilibrium shear flow.
+
+    "Florian Mueller-Plathe. Reversing the perturbation in nonequilibrium
+     molecular dynamics: An easy way to calculate the shear viscosity of fluids.
+     Phys. Rev. E, 59:4894-4898, May 1999."
+
+    The simulation box is divided in a number of slabs.  Two distinct slabs of
+    those are chosen. The "max" slab searches for the maximum velocity component
+    in flow direction while the "min" slab searches for the minimum velocity
+    component. Afterward, both velocity components are swapped.
+
+    This introduces a momentum flow, which drives the flow. The strength of this
+    flow is set through the `flow_target` argument, which defines a target value
+    for the time-integrated momentum flux. The searching and swapping is
+    repeated until the target is reached. Depending on the target sign, the
+    "max" and "min" slab might be swapped.
+
+    Args:
+        trigger: trigger (hoomd.trigger.trigger_like): Select the time steps
+        on which to sort.
+
+        num_swaps: maximum number of swaps
+
+        slab_wdith: thickness of slabs
+
+        slab_separation: distance between slabs
+
+        target_momentum: target momentum for particles in the slabs
```

Comment:
I am wondering if we would make this have a default value of `None`, which is translated into a very large value internally. That way, the algorithm behaves like the normal Muller-Plathe method if this is not set.

Reply:
I think that should be better